exercises/zipper/example.js

```diff
@@ -1,104 +1,100 @@
 'use strict';
 
-var Zipper = (function () {
-  function Zipper(tree, trail) {
-    this.tree = tree;
-    this.trail = trail;
-  }
-
-  Zipper.fromTree = function (tree) {
-    return new Zipper(tree, []);
-  };
-
-  function fromTrail(tree, last) {
-    if (last[0] === 'left') {
-      return {
-        value: last[1],
-        left: tree,
-        right: last[2]
-      };
-    }
+function fromTrail(tree, last) {
+  if (last[0] === 'left') {
     return {
       value: last[1],
-      left: last[2],
-      right: tree
+      left: tree,
+      right: last[2]
     };
   }
-
-  function rebuildTree(tree, trail) {
-    if (trail.length === 0) return tree;
-
-    var last = trail[0];
-    return rebuildTree(fromTrail(tree, last), trail.slice(1));
-  }
-
-  Zipper.prototype.toTree = function () {
-    return rebuildTree(this.tree, this.trail);
-  };
-
-  Zipper.prototype.value = function () {
-    return this.tree.value;
-  };
-
-  Zipper.prototype.left = function () {
-    if (!this.tree.left) return null;
-
-    return new Zipper(
-      this.tree.left,
-      [['left', this.tree.value, this.tree.right]].concat(this.trail)
-    );
+  return {
+    value: last[1],
+    left: last[2],
+    right: tree
   };
-
-  Zipper.prototype.right = function () {
-    if (!this.tree.right) return null;
-
-    return new Zipper(
-      this.tree.right,
-      [['right', this.tree.value, this.tree.left]].concat(this.trail)
-    );
-  };
-
-  Zipper.prototype.up = function () {
-    if (this.trail.length === 0) return null;
-
-    var last = this.trail[0];
-    return new Zipper(fromTrail(this.tree, last), this.trail.slice(1));
-  };
-
-  Zipper.prototype.setValue = function (value) {
-    return new Zipper(
-      {
-        value: value,
-        left: this.tree.left,
-        right: this.tree.right
-      },
-      this.trail
-    );
-  };
-
-  Zipper.prototype.setLeft = function (left) {
-    return new Zipper(
-      {
-        value: this.tree.value,
-        left: left,
-        right: this.tree.right
-      },
-      this.trail
-    );
-  };
-
-  Zipper.prototype.setRight = function (right) {
-    return new Zipper(
-      {
-        value: this.tree.value,
-        left: this.tree.left,
-        right: right
-      },
-      this.trail
-    );
-  };
-
-  return Zipper;
-})();
+}
+
+function rebuildTree(tree, trail) {
+  if (trail.length === 0) return tree;
+
+  var last = trail[0];
+  return rebuildTree(fromTrail(tree, last), trail.slice(1));
+}
+
+var Zipper = function (tree, trail) {
+  this.tree = tree;
+  this.trail = trail;
+};
+
+Zipper.fromTree = function (tree) {
```

Comment:
Should this be declared on the prototype, like all the other functions?  Or is this a deliberate choice?

Reply:
This is meant to be a "constructor", like `Array.from()`. That means the following should not work:
```js
zipper.left().right().fromTree(someTree)
```

Does that make sense?

Reply:
The `README` describes `fromTree` at the same level as `toTree`,... But it makes sense to have it like a kind of a named constructor. So, I like `Zipper.fromTree` and the following doesn't make sense for me:

```javascript
zipper.left().right().fromTree(someTree)
```

because, is it creating a tree from `someTree` or from `zipper.left().right()`?

Reply:
Yes, @rchavarria, exactly what I meant :)